libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/location/replay/ReplayRouteLocationEngine.java

```diff
@@ -0,0 +1,168 @@
+package com.mapbox.services.android.navigation.v5.location.replay;
+
+import android.annotation.SuppressLint;
+import android.location.Location;
+import android.os.Handler;
+import android.support.annotation.Nullable;
+
+import com.mapbox.android.core.location.LocationEngine;
+import com.mapbox.android.core.location.LocationEngineListener;
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.geojson.LineString;
+import com.mapbox.geojson.Point;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class ReplayRouteLocationEngine extends LocationEngine implements Runnable {
+
+  private static final int HEAD = 0;
+  private ReplayRouteLocationConverter converter;
+  private Handler handler;
+  private List<Location> mockedLocations;
+  private ReplayLocationDispatcher dispatcher;
+  private Location lastLocation = new Location("ReplayRouteLocation");
+  private final ReplayLocationListener replayLocationListener = new ReplayLocationListener() {
```

Comment:
I've tried what you suggested but this would need a bigger refactor especially because we're referencing some properties within the listener that would need to be accessed / shared properly i.e. `locationListeners` from `LocationEngine`, `lastLocation` and `mockedLocations`.

For now, I'd say leave it as is and revisit later if it becomes an issue. What do you think?

Reply:
@Guardiola31337 sounds good